druid/src/widget/list.rs

```diff
@@ -34,6 +34,126 @@ use crate::{
 };
 
 /// A list widget for a variable-size collection of items.
+/// For a static sized collection of items use [`Flex`].
+/// The best data structure to use with the list widget is [`Vector`] from the [`im`] feature.
+/// To include the [`im`] feature, update Cargo.toml
+/// ```ignore
+/// [dependencies]
+/// // before
+/// druid = "0.7.0"
+/// // after
+/// druid = { version = "0.7.0", features = ["im"] }
+/// ```
+///
+/// # Examples
+///
+/// ## Define the data
+///
+/// ```
+/// # use druid::{ Data, Lens };
+/// use druid::im::Vector;
+/// #[derive(Clone, Data, Lens)]
+/// struct AppState {
+///     list_data: Vector<String>,
+/// }
+/// ```
+/// ## Create initial State
+///
+/// ```
+/// # use druid::{ Data, Lens};
+/// # use druid::im::Vector;
+/// # #[derive(Clone, Data, Lens)]
+/// # struct AppState {
+/// #     list_data: Vector<String>,
+/// # }
+/// let initial_state = AppState {
+///     list_data: Vector::from(
+///         vec!(
+///             "one".into(),
+///             "two".into(),
+///             "three".into(),
+///             "four".into(),
+///         )),
+/// };
+/// ```
+/// ## Create widgets
+///
+/// ```
+/// # use druid::widget::{ Label, List };
+/// # use druid::{ Data, Lens, Widget, WidgetExt, Env};
+/// # use druid::im::Vector;
+/// # #[derive(Clone, Data, Lens)]
+/// # struct AppState {
+/// #     list_data: Vector<String>,
+/// # }
+/// fn root() -> impl Widget<AppState> {
+///     let list = List::new(list_item)
+///         .lens(AppState::list_data);
+///     list
+/// }
+///
+/// fn list_item() -> impl Widget<String> {
+///     let label = Label::new(|data: &String, _env: &Env|
+///         data.clone());
+///     label
+/// }
+/// ```
```

Comment:
This whole example seems way too big to me. It's the example equivalent of an integration test, where you'd want a unit test.

I'd recommend skipping the `AppState` data structure and the lensing, and just using `Vector<String>` in your example.